capabilities/writetarget/beholder/pb/platform/error.go

```diff
@@ -0,0 +1,13 @@
+package writetarget
+
+import (
+	"fmt"
+
+	protoimpl "google.golang.org/protobuf/runtime/protoimpl"
+)
+
+// AsError returns the WriteError message as an (Go) error
+func (e *WriteError) Error() error {
+	protoName := protoimpl.X.MessageTypeOf(e).Descriptor().FullName()
+	return fmt.Errorf("%s [ERR-%v] - %s: %s", protoName, e.Code, e.Summary, e.Cause)
```

Comment:
I think this is confusing. If you need to _return_ an `error`, then IMHO `AsError()` made more sense. My suggestion was to _implement_ the error interface:
```suggestion
func (e *WriteError) Error() string {
	protoName := protoimpl.X.MessageTypeOf(e).Descriptor().FullName()
	return fmt.Sprintf("%s [ERR-%v] - %s: %s", protoName, e.Code, e.Summary, e.Cause)
```
Then users don't have to call a method at all.

Reply:
I'm not really sure how implementing the `error` interface is helpful here, since the user of this function [asEmittedError](https://github.com/smartcontractkit/chainlink-framework/blob/9c761e0d44ba9ae8878dff01dd2d6e91ee41012c/capabilities/writetarget/write_target.go#L471-L479) needs `AsError` to return an error?

Reply:
It would already be an error, so they would not need to call the method to convert it.

Reply:
ah ok, I see your point. The brain isn't working great after a long day, I just updated it